web_poet/fields.py

```diff
@@ -73,33 +88,55 @@ def __get__(self, instance, owner=None):
         return _field
 
 
-async def item_from_fields(obj, item_cls=dict, *, item_cls_fields=False):
+def get_fields_dict(cls_or_instance) -> Dict[str, FieldInfo]:
+    """Return a dictionary with information about the fields defined
+    for the class: keys are field names, and values are
+    :class:`web_poet.fields.FieldInfo` instances.
+    """
+    return getattr(cls_or_instance, _FIELDS_INFO_ATTRIBUTE_READ, {})
+
+
+T = TypeVar("T")
+
+
+# FIXME: type is ignored as a workaround for https://github.com/python/mypy/issues/3737
+# inference works properly if a non-default item_cls is passed; for dict
+# it's not working (return type is Any)
+async def item_from_fields(
+    obj, item_cls: Type[T] = dict, *, skip_nonitem_fields: bool = False  # type: ignore[assignment]
+) -> T:
     """Return an item of ``item_cls`` type, with its attributes populated
     from the ``obj`` methods decorated with :class:`field` decorator.
 
-    If ``item_cls_fields`` is True, ``@fields`` whose names don't match
-    any of the ``item_cls`` attributes are not passed to ``item_cls.__init__``.
-    When ``item_cls_fields`` is False (default), all ``@fields`` are passed
-    to ``item_cls.__init__``.
+    If ``skip_nonitem_fields`` is True, ``@fields`` whose names are not
```

Comment:
I've renamed item_cls_fields to skip_nonitem_fields in this PR; does it look ok?